clients/filesystem-hadoop3/src/main/java/org/apache/gravitino/filesystem/hadoop/GravitinoVirtualFileSystem.java

```diff
@@ -559,6 +563,19 @@ public long getDefaultBlockSize(Path f) {
     return context.getFileSystem().getDefaultBlockSize(context.getActualFileLocation());
   }
 
+  @Override
+  public Token<?>[] addDelegationTokens(String renewer, Credentials credentials) {
+    List<Token<?>> tokenList = Lists.newArrayList();
+    for (FileSystem fileSystem : internalFileSystemCache.asMap().values()) {
```

Comment:
Shall we filter out some filesystems that doesn't have delegation token mechanism?

Reply:
I think we don't need to do filtering, please see the specific code call:
![image](https://github.com/user-attachments/assets/9097751d-e12e-4465-a79b-0b8f4cce92d4)
![image](https://github.com/user-attachments/assets/cbc0e323-77cf-4fb4-8006-bdbdd645e60a)
![image](https://github.com/user-attachments/assets/9c681db1-c372-4a27-b8aa-324caa2d51a8)
![image](https://github.com/user-attachments/assets/c3f0c147-5f0d-47bc-8690-e81c44ddcde7)
1. The `getCanonicalServiceName` and `getChildFileSystems` methods will always return `false` in the default implementations.
2. Some FileSystems do not support token will also override the `getCanonicalServiceName` to return `null` directly(like S3AFileSystem, AliyunOSSFileSystem)
![image](https://github.com/user-attachments/assets/fd8cbc52-a6eb-4ead-b4cc-b56d53f298ab)

Reply:
I see, thanks for the explanation.